build-logic/src/main/kotlin/PublicationPlugin.kt

```diff
@@ -134,10 +134,17 @@ abstract class PublicationPlugin : Plugin<Project> {
                 // See https://docs.gradle.org/current/userguide/signing_plugin.html#sec:using_gpg_agent how to configure it
                 // useGpgCmd()
 
-                val signingKeyId = System.getenv("ORG_GRADLE_PROJECT_signingKeyId")
-                val signingKey = System.getenv("ORG_GRADLE_PROJECT_signingKey")
-                val signingPassword = System.getenv("ORG_GRADLE_PROJECT_signingKeyPassword")
-                useInMemoryPgpKeys(signingKeyId, signingKey, signingPassword)
+                val signingKeyId = System.getenv("ORG_GRADLE_PROJECT_signingKeyId")?.takeIf { it.isNotEmpty() }
+                val signingKey = System.getenv("ORG_GRADLE_PROJECT_signingKey")?.takeIf { it.isNotEmpty() }
+                val signingPassword = System.getenv("ORG_GRADLE_PROJECT_signingKeyPassword")?.takeIf { it.isNotEmpty() }
+                if (signingKeyId != null && signingKey != null && signingPassword != null) {
+                    // Avoid setting empty strings as signing keys. This avoids breaking the build when PR is opened from a fork.
+                    // Also, due to https://github.com/gradle/gradle/issues/18477 signing tasks try to prematurely access the signatory.
+                    // This also improves error messages if something's misconfigured
+                    useInMemoryPgpKeys(signingKeyId, signingKey, signingPassword)
```

Comment:
The new task I added (`signShadowJarExecutable`) gets created eagerly, tries to access the signatory and fails if the signing info contains empty string (not nulls) 

As a workaround I'm proposing to not configure the signing. This solves the misbehavior of the Sign task but also should not fail PRs from froks created _during_ release (the `isRequired` property is not the only check that determines if the signing tasks should run - [link](https://github.com/gradle/gradle/blob/c5118d30e4d4996aa1612ff584e68469310f9a3c/platforms/software/signing/src/main/java/org/gradle/plugins/signing/Sign.java#L100))